File: test/e2e/test-files/workspaces/long-running-loop/loop.r
loop_number <- 0
duration_seconds <- 180
start_time <- Sys.time()

cat(sprintf("Starting loop for %d seconds...\n", duration_seconds))

while (TRUE) {
  current_time <- Sys.time()
  elapsed_time <- as.numeric(difftime(current_time, start_time, units = "secs"))

  if (elapsed_time >= duration_seconds) {
    cat(sprintf(
      "Completed %d iterations over %d seconds, exiting.\n",
      loop_number,
      duration_seconds
    ))
    break
  }

  cat(sprintf("Loop: %d\n", loop_number))
  loop_number <- loop_number + 1
  Sys.sleep(1) # Sleep for 1 second
}

File: test/e2e/test-files/workspaces/long-running-loop/loop.py
import time

loop_number = 0
duration_seconds = 180
start_time = time.time()

print(f"Starting loop for {duration_seconds} seconds...")

while True:
    current_time = time.time()
    elapsed_time = current_time - start_time

    if elapsed_time >= duration_seconds:
        print(f"Completed {loop_number} iterations over {duration_seconds} seconds, exiting.")
        break
    
    print(f"Loop: {loop_number}")
    loop_number += 1
    time.sleep(1)  # Sleep for 1 second
